web/src/lib/domain-logic/allowFormation.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  BusinessPersona,
+  FormationLegalType,
+  PublicFilingLegalType,
+  publicFilingLegalTypes,
+} from "@businessnjgovnavigator/shared";
+
+export const castPublicFilingLegalTypeToFormationType = (
+  legalStructureId: PublicFilingLegalType,
+  persona: BusinessPersona | undefined
+): FormationLegalType => {
+  return `${persona == "FOREIGN" ? "foreign-" : ""}${legalStructureId}` as FormationLegalType;
+};
+
+export const allowFormation = (
+  legalStructureId: string | undefined,
+  persona: BusinessPersona | undefined
+) => {
+  const featureFlagMap: Partial<Record<FormationLegalType, boolean>> = {
+    "foreign-limited-liability-company": process.env.FEATURE_BUSINESS_FLC == "true",
+    "foreign-limited-liability-partnership": process.env.FEATURE_BUSINESS_FLLP == "true",
+    "foreign-limited-partnership": process.env.FEATURE_BUSINESS_FLP == "true",
+    "foreign-s-corporation": process.env.FEATURE_BUSINESS_FCORP == "true",
+    "foreign-c-corporation": process.env.FEATURE_BUSINESS_FCORP == "true",
+  };
+  if (publicFilingLegalTypes.includes(legalStructureId as PublicFilingLegalType)) {
+    return (
+      featureFlagMap[
+        castPublicFilingLegalTypeToFormationType(legalStructureId as PublicFilingLegalType, persona)
+      ] ?? true
+    );
+  }
+  return false;
```

Comment:
I think this file is a little hard to read... do you think the logic here could be simplified a bit?

If I were to describe this aloud to someone, I'd say something like:
- if Trade Name, return false always
- otherwise, if not foreign, return true always
- otherwise, return based on whether the feature flag is enabled

it seems like that could be translated more directly into code without a lot of the layers of abstraction and re-structuring and re-direction here that makes it harder for someone to look at the code and understand the logic driving it